rpc/lib/server/handlers.go

```diff
@@ -204,13 +210,14 @@ func arrayParamsToArgs(rpcFunc *RPCFunc, cdc *amino.Codec, params []json.RawMess
 	return values, nil
 }
 
-// `raw` is unparsed json (from json.RawMessage) encoding either a map or an array.
-// `argsOffset` should be 0 for RPC calls, and 1 for WS requests, where len(rpcFunc.args) != len(rpcFunc.argNames).
+// raw is unparsed json (from json.RawMessage) encoding either a map or an
+// array.
 //
 // Example:
-//   rpcFunc.args = [rpctypes.WSRPCContext string]
+//   rpcFunc.args = [rpctypes.Context string]
 //   rpcFunc.argNames = ["arg"]
-func jsonParamsToArgs(rpcFunc *RPCFunc, cdc *amino.Codec, raw []byte, argsOffset int) ([]reflect.Value, error) {
+func jsonParamsToArgs(rpcFunc *RPCFunc, cdc *amino.Codec, raw []byte) ([]reflect.Value, error) {
+	const argsOffset = 1
```

Comment:
I wonder if it was really worth adding the Context to every function or if we really only needed to add it to a select few - ie. replace it in the Subscrube/Unsubscribe and add it to the BroadcastTxCommit. Then we'd just be increasing the list of methods that have `argsOffset = 1`. 

Is there some reason we really need it on all the other methods?

Reply:
The code can be further simplified now that we have context everywhere.

> Is there some reason we really need it on all the other methods?

Other - no. Subscribe/Unsubscribe/UnsubscribeAll/Broadcast - yes. Note the latter group can include future methods.

But having some methods with context and others without will be super confusing for developers and inconsistent.